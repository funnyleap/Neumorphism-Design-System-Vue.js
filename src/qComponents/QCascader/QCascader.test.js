import Component from './src/QCascader';

describe('QCascader', () => {
  let instance;
  let options;

  beforeEach(() => {
    options = {
      mocks: {
        $t: () => {}
      },
      propsData: {
        value: 'resource',
        placeholder: 'placeholder',
        options: [
          {
            value: 'guide',
            label: 'Guide',
            children: [
              {
                value: 'child',
                label: 'Child',
                children: [{ value: 'next child', label: 'Next child' }]
              }
            ]
          },
          {
            value: 'resource',
            label: 'Resource'
          }
        ]
      }
    };

    instance = mount(Component, options);
  });

  it('should match snapshot', () => {
    expect(instance.element).toMatchSnapshot();
  });

  it('data should match snapshot', () => {
    expect(Component.data()).toMatchSnapshot();
  });

  describe('computed', () => {
    describe('model', () => {
      it('should set inputValue if has been changed', () => {
        const expected = 'guide';
        instance.vm.model = expected;

        expect(instance.vm.inputValue).toEqual(expected);
      });
    });
    describe('clearBtnVisible', () => {
      it('should return false if showClose is false', () => {
        instance.setData({
          showClose: false
        });
        expect(instance.vm.clearBtnVisible).toBeFalsy();
      });
      it('should return true if showClose is true', () => {
        instance.setData({
          showClose: true
        });
        expect(instance.vm.clearBtnVisible).toBeTruthy();
      });
    });
  });

  describe('watch', () => {
    describe('value', () => {
      it('should set checkedValues if value is array', async () => {
        const expected = ['guide', 'child'];
        await instance.setProps({
          value: expected
        });
        expect(instance.vm.checkedValues).toEqual(expected);
      });
    });
  });
});
