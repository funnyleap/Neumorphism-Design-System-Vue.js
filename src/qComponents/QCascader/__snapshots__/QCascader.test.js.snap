// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QCascader data should match snapshot 1`] = `
Object {
  "areTagsHovered": false,
  "checkedNodes": Array [],
  "checkedValues": Array [],
  "inputInitialHeight": 0,
  "inputValue": null,
  "popper": null,
  "presentText": null,
  "showClose": false,
}
`;

exports[`QCascader should match snapshot 1`] = `
<div
  class="q-cascader"
>
  <div
    class="q-input q-input_suffix"
  >
    <!---->
     
    <input
      aria-label=""
      autocomplete="off"
      class="q-input__inner"
      placeholder="placeholder"
      readonly="readonly"
      tabindex=""
      type="text"
    />
     
    <span
      class="q-input__suffix"
    >
      <span
        class="q-input__suffix-inner"
      >
        <span
          class="q-input__icon q-icon-triangle-down"
        />
         
        <!---->
         
        <!---->
      </span>
    </span>
  </div>
   
  <!---->
   
  <div
    class="q-cascader-panel"
  >
    <div
      class="q-cascader-menu"
    >
      <div
        class="q-cascader-menu__wrap"
      >
        <div
          class="q-scrollbar"
        >
          <div
            class="q-scrollbar__wrap q-cascader-menu__scrollbar q-scrollbar__wrap_hidden-default"
          >
            <div
              class="q-scrollbar__view"
            >
              <div
                class="q-cascader-node"
              >
                <!---->
                 
                <span
                  class="q-cascader-node__label"
                >
                  Guide
                </span>
                 
                <span
                  class="q-cascader-node__postfix q-icon-triangle-right"
                />
              </div>
              <div
                class="q-cascader-node"
              >
                <!---->
                 
                <span
                  class="q-cascader-node__label"
                >
                  Resource
                </span>
                 
                <!---->
              </div>
            </div>
          </div>
           
          <div
            class="q-scrollbar__bar q-scrollbar__bar_horizontal"
            style="display: none;"
          >
            <div
              class="q-scrollbar__thumb"
              style="transform: translateX(0%); webkit-transform: translateX(0%);"
            />
          </div>
           
          <div
            class="q-scrollbar__bar q-scrollbar__bar_vertical"
            style="display: none;"
          >
            <div
              class="q-scrollbar__thumb"
              style="transform: translateY(0%); webkit-transform: translateY(0%);"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
